### ISSUE TITLE
Group 18 - fxtracker (py)

### ISSUE THREAD

**Issue by markusnam** [UBC-MDS/software-review-2023#19]
Submitting Author: Sarah Abdelazim (@missarah96), Lennon Au-Yeung (@lennonay), Crystal Geng (@THF-d8), Markus Nam (@markusnam)
All current maintainers: (@missarah96, @lennonay, @THF-d8, @markusnam)
Package Name: fxtracker
One-Line Description of Package: This package allows user to perform currency conversion, target rate lookup and plotting of rate history and profit/loss percentage history.
Repository Link: https://github.com/UBC-MDS/fxtracker
Version submitted: v1.0.0
Editor: Sarah Abdelazim (@missarah96), Lennon Au-Yeung (@lennonay), Crystal Geng (@THF-d8), Markus Nam (@markusnam)  
Reviewer 1: Austin Shih  
Reviewer 2: Andy Wang
Reviewer 3: Bruce Wu
Reviewer 4: Shirley Zhang  
Archive: TBD  
Version accepted: TBD 
Date accepted (month/day/year): TBD

---

## Description

This is a package created as a group project for DSCI_524 Collaborative Software Development of UBC Master of Data Science (MDS) program 2022-2023. Based on the foreign exchange data in Yahoo Finance, this package allows user to perform currency conversion based on the latest available exchange rate, lookup a target exchange rate from historical data as well plotting exchange rate history and profit/loss percentage history by specifying a currency pair (and other input parameters).

## Scope 
- Please indicate which [category or categories][PackageCategories] this package falls under:
	- [x] Data retrieval
	- [ ] Data extraction
	- [ ] Data munging
	- [ ] Data deposition
	- [ ] Reproducibility
	- [ ] Geospatial
	- [ ] Education
	- [x] Data visualization*

> *Please fill out a pre-submission inquiry before submitting a data visualization package. For more info, see [notes on categories][NotesOnCategories] of our guidebook.*

- **For all submissions**, explain how the and why the package falls under the categories you indicated above. In your explanation, please address the following points (briefly, 1-2 sentences for each):  

  - Who is the target audience and what are scientific applications of this package?
  The target audience is anyone who is interested in the foreign exchange market. e.g. travellers, professional investors, financial institutions.

  - Are there other Python packages that accomplish the same thing? If so, how does yours differ?
  There is a python package ([forex-python](https://pypi.org/project/forex-python/)) relevant to foreign exchange. That package is basically for retrieving exchange rates and bitcoin prices in plain text as well as performing conversion. It does not provide visualizations and lookup function like `fxtracker` does. `fxtracker` allows user to visualize the trends and understand if a target price of a currency pair of interest is within a reasonable range.

  - If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or `@tag` the editor you contacted:

## Technical checks

For details about the pyOpenSci packaging requirements, see our [packaging guide][PackagingGuide]. Confirm each of the following by checking the box.  This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] has an [OSI approved license][OsiApprovedLicense].
- [x] contains a README with instructions for installing the development version. 
- [x] includes documentation with examples for all functions.
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a test suite.
- [x] has continuous integration, such as Travis CI, AppVeyor, CircleCI, and/or others.

## Publication options

- [ ] Do you wish to automatically submit to the [Journal of Open Source Software][JournalOfOpenSourceSoftware]? If so:

<details>
 <summary>JOSS Checks</summary>  

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements][JossSubmissionRequirements]. Be aware that completing the pyOpenSci review process **does not** guarantee acceptance to JOSS. Be sure to read their submission requirements (linked above) if you are interested in submitting to JOSS.
- [ ] The package is not a "minor utility" as defined by JOSS's [submission requirements][JossSubmissionRequirements]: "Minor ‘utility’ packages, including ‘thin’ API clients, are not acceptable." pyOpenSci welcomes these packages under "Data Retrieval", but JOSS has slightly different criteria.
- [ ] The package contains a `paper.md` matching [JOSS's requirements][JossPaperRequirements] with a high-level description in the package root or in `inst/`.
- [ ] The package is deposited in a long-term repository with the DOI: 

*Note: Do not submit your package separately to JOSS*
  
</details>

## Are you OK with Reviewers Submitting Issues and/or pull requests to your Repo Directly?
This option will allow reviewers to open smaller issues that can then be linked to PR's rather than submitting a more dense text based review. It will also allow you to demonstrate addressing the issue via PR links.

- [x] Yes I am OK with reviewers submitting requested changes as issues to my repo. Reviewers will then link to the issues in their submitted review.

## Code of conduct

- [x] I agree to abide by [pyOpenSci's Code of Conduct][PyOpenSciCodeOfConduct] during the review process and in maintaining my package should it be accepted.

## Please fill out our survey
- [ ] [Last but not least please fill out our pre-review survey](https://forms.gle/F9mou7S3jhe8DMJ16). This helps us track
submission and improve our peer review process. We will also ask our reviewers 
and editors to fill this out.

**P.S.** *Have feedback/comments about our review process? Leave a comment [here][Comments]

## Editor and Review Templates

The [editor template can be found here][Editor Template].

The [review template can be found here][Review Template].

[PackagingGuide]: https://www.pyopensci.org/contributing-guide/authoring/index.html#packaging-guide

[PackageCategories]: https://www.pyopensci.org/contributing-guide/open-source-software-peer-review/aims-and-scope.html?highlight=data#package-categories

[NotesOnCategories]: https://www.pyopensci.org/contributing-guide/open-source-software-peer-review/aims-and-scope.html?highlight=data#notes-on-categories

[JournalOfOpenSourceSoftware]: http://joss.theoj.org/

[JossSubmissionRequirements]: https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements

[JossPaperRequirements]: https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain

[PyOpenSciCodeOfConduct]: https://www.pyopensci.org/contributing-guide/open-source-software-peer-review/code-of-conduct.html?highlight=code%20conduct

[OsiApprovedLicense]: https://opensource.org/licenses

[Editor Template]: https://www.pyopensci.org/peer-review-guide/software-peer-review-guide/editors-guide.html#respond-to-the-submitter-in-the-github-issue

[Review Template]: https://www.pyopensci.org/peer-review-guide/software-peer-review-guide/reviewer-guide.html#peer-review-template

[Comments]: https://github.com/pyOpenSci/governance/issues/8

**Comment by tiger12055** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [ ] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [x] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [x] A [repostatus.org](https://www.repostatus.org/) badge,
    - [x] Python versions supported,
    - [x] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [ ] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [x] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing:

---

#### Review Comments
1. It would be helpful to include your email and a link to your GitHub account in the contribution file so that others can easily get in touch with you regarding your contributions
2. It is recommended to organize the functions into separate files based on their primary functionality. This will make it easier for readers to understand the main purpose of each function and also aid in writing unit tests. Grouping visualization functions together and other functions separately can enhance the readability and maintainability of the code.
3. It would be helpful to include a brief introduction to the yfinance package and the datasets it includes, as it serves as the foundation for your package. This would allow users with limited background knowledge to better understand the context of the data and the purpose of your package.
4. The documentation of your package is excellent and user-friendly. I was able to run the package smoothly without encountering any difficulties. Your group has paid close attention to every detail, making the user experience seamless. Great job!
5. 100% test coverage. Great job

**Comment by shlrley** [comment]
**Reviewer:** Shirley Zhang

## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [ ] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [ ] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [ ] Python versions supported,
    - [ ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [ ] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1.5 hours 

---

#### Review Comments

This is a very interesting and useful package and I enjoyed going through how your group implemented all the functions. Overall it has been built very well, everything appears to have run smoothly on my end, and it has been well documented. Here are some things you guys could possibly do to improve the package: 

1. Include citation information to the Yahoo Finance foreign exchange data, as well as any other sources you may have used to create the functions. 

2. Set up the repository status badge [repostatus.org](https://www.repostatus.org/), the Python versions supported badge, and the current package version (on PyPI / Conda) badge as listed in the above package review checklist. 

3. In the README.md documentation, the wording for the description of the function [`fx_rate_lookup`](https://github.com/UBC-MDS/fxtracker/blob/main/README.md#to-look-up-the-first-date-reverse-chronological-order-on-which-the-target-price-falling-between-day-high-and-day-low-based-on-the-availability-of-data) is a bit confusing. Perhaps "the first date (reverse chronological order)" could be simplified to something like "the latest date" or "the most recent date"? I also did not fully understand the meaning of "the target price falling between day high and day low" - could it be reworded? 

4. Inside the [`example.ipynb`](https://github.com/UBC-MDS/fxtracker/blob/main/docs/example.ipynb) file, the graphs created after `price_trend_viz` cannot be seen. Since this function opens up a new window (hence the graph isn't shown), perhaps it would be nice to create a new markdown cell underneath and just include a screenshot of what the graph is supposed to look like. 

5. (optional) Currently, the function `price_trend_viz` will output the created graphs in a new browser window with Altair. Is it possible to just have the graph show up (either as a .png or another format) directly onto the screen (i.e if you're running on jupyter notebook) or in a new terminal pop-up window (if you're running on terminal)? 

6. (optional) Building off of 4, it would be nice to also save the plot as a .png or image file once it's been created - perhaps as a future function you guys could implement? 

Overall, congratulations on this Python package! It was well-designed and easy to follow. Would love to see how you guys continue improving it!

**Comment by austin-shih** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [x] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [ ] Python versions supported,
    - [ ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [ ] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 20 minutes

---

#### Review Comments

- Very nicely done! I can see myself using this in the future. The functions are easy to use and self-explanatory. I like how I can visualize the currency changes over time and that I can do this with every currency available on the reference website.
- Good job! This was fun to play around with!

**Comment by BruceUBC** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [ ] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [x] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [x] A [repostatus.org](https://www.repostatus.org/) badge,
    - [x] Python versions supported,
    - [x ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [ ] Link to your documentation website.
- [ ] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [ ] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1 hour

---

#### Review Comments
1.  The documentation and usage of the package are entirely constructed with detailed instructions, which helps users install and test the functions easily. Some possible improvements could be adding some description to the plot in the usage part.  
2.  I have read the discussion of license choice in your issues. The reason why choosing an MIT license is fully explained in detailed grounds. Choosing an appropriate license is important for the python package. Your choice is trustworthy for users.  
3.  Functions are clearly written in the script with perfect structure and logic. Some certain comments could be further explained such as "check the input of curr1".  It would be convenient for users to understand the meaning of the function with detailed comments especially when there are several conditions in the function.
4.  When introducing the input of the functions, the possible range of the input could be included in the description. For example, when plotting the exchange rate history, what is the possible time period based on your chosen dataset?  
5.  It is smart to use branches to equally distributed tasks and review any pull request from each other. Your teamwork looks really good!  
In a nutshell, thank you so much for creating such a useful package. I am looking forward to seeing any improvement in your package and using some of the functions in the future. Great work! It is a pleasure to read your package.
